src/engine/Server.cpp

```diff
@@ -742,7 +754,14 @@ CPP_template_def(typename RequestT)(
   std::string_view acceptHeader = request.base()[http::field::accept];
 
   if (!mediaType.has_value()) {
-    mediaType = ad_utility::getMediaTypeFromAcceptHeader(acceptHeader);
+    try {
+      mediaType = ad_utility::getMediaTypeFromAcceptHeader(acceptHeader);
```

Comment:
Cant the `getMediaTypeFromAcceptHeader`  throw `HttpError(not_acceptable)` directly?
(I think i commented this also above in the implementation)

Reply:
Partly because I didn't think about the correct behaviour for Query/Update in that case and partly because I didn't want to open the can of worms which is the MediaType code. I think that it needs some love. We already discussed that it can be improved.